rfcs/text/0033-tty-output.md

```diff
@@ -0,0 +1,157 @@
+# 0000: Process IO events
+<!-- Leave this ID at 0000. The ECS team will assign a unique, contiguous RFC number upon merging the initial stage of this RFC. -->
+
+- Stage: **0 (strawperson)** <!-- Update to reflect target stage. See https://elastic.github.io/ecs/stages.html -->
+- Date: **TBD** <!-- The ECS team sets this date at merge time. This is the date of the latest stage advancement. -->
+
+<!--
+As you work on your RFC, use the "Stage N" comments to guide you in what you should focus on, for the stage you're targeting.
+Feel free to remove these comments as you go along.
+-->
+
+<!--
+Stage 0: Provide a high level summary of the premise of these changes. Briefly describe the nature, purpose, and impact of the changes. ~2-5 sentences.
+-->
+
+The goal of the RFC is to introduce a new fieldset (IO) to track input and output data from processes. The initial implementation will be focused on capturing TTY output from processes writing to either stdout or stderr (FDs 1 and 2). Each event will contain many lines of tty output (configurable), and will have some of its process fields populated to help associate the IO data with the process that is generating it (as well as the its associated session).
+
+<!--
+Stage 1: If the changes include field additions or modifications, please create a folder titled as the RFC number under rfcs/text/. This will be where proposed schema changes as standalone YAML files or extended example mappings and larger source documents will go as the RFC is iterated upon.
+-->
+
+see: 0033/io.yml
+
+<!--
+Stage X: Provide a brief explanation of why the proposal is being marked as abandoned. This is useful context for anyone revisiting this proposal or considering similar changes later on.
+-->
+
+## Fields
+
+<!--
+Stage 1: Describe at a high level how this change affects fields. Include new or updated yml field definitions for all of the essential fields in this draft. While not exhaustive, the fields documented here should be comprehensive enough to deeply evaluate the technical considerations of this change. The goal here is to validate the technical details for all essential fields and to provide a basis for adding experimental field definitions to the schema. Use GitHub code blocks with yml syntax formatting, and add them to the corresponding RFC folder.
+-->
+
+process.io (nested IO fieldset)
```

Comment:
nit: Markdown's acting silly and rendering all the fields in one line.

`process.io (nested IO fieldset) process.io.data (array of objects) process.io.data.time (date) process.io.data.value (text)`

Reply:
ah yea, i'll do a review in preview mode. thanks.